### ISSUE TITLE
Question: Parallel processing and EAGLE-RC

### ISSUE THREAD

**Issue by kenji-yt** [tony-kuo/eagle#9]
Hi,

Is there is a possibility to run EAGLE-RC on multiple cores? 

If not, does that mean it will use only a single core, or how many can we expect it to use? 

Thank you in advance,

Kenji

**Comment by tony-kuo** [comment]
There are 2 workflows:
1. EAGLE is run first then EAGLE-RC parses the results. The EAGLE part is multi-threaded and can use multiple cores
2. The --ngi parameter is used in EAGLE-RC. This unfortunately is single threaded.

**Comment by kenji-yt** [comment]
Thanks a lot for your reply. 

To confirm, this mean that if I run using the --ngi flag, only 1 thread will be used by the entire process?

**Comment by tony-kuo** [comment]
Correct. Hopefully it shouldn't be too slow as it calculates the likelihood of every read in the bam against the reference. IO should be the bottleneck in this case in my estimation.

**Comment by kenji-yt** [comment]
Great, thanks! I also have a question regarding the --ngi mode. If I understand correctly, with --ngi we can only sort reads between two genomes? I wanted to use it for any number of genomes but it seems not to work.

**Comment by tony-kuo** [comment]
Hello, in order to do more than two genomes, you need to do multiple pair-wise comparisons.
For example in hexaploids, you can run A vs B, A vs C, B vs C. Then use the outputs from that to build a consensus.

This is shown in the existing hexaploid [script](https://github.com/tony-kuo/eagle/blob/master/scripts/hexaploid_ngi_analysis.sh)

If you need higher ploidy, you'll have to work out the combinations of pair-wise comparisons and calculate the probabilities using the current scripts as templates.

**Comment by kenji-yt** [comment]
Hi, thank you very much for your help! 
I would like to ask a few more questions if that's ok:

- 1: [The script](https://github.com/tony-kuo/eagle/blob/master/scripts/hexaploid_ngi_analysis.sh) you referred to uses a python script called "ref_ngi_consensus.py". I cannot find this exact script, but I believe it is the same as the one called "[ref3_ngi_consensus.py](https://github.com/tony-kuo/eagle/blob/master/scripts/ref3_ngi_consensus.py)". Is that correct? 

- 2: You "recommend that the chromosome names are different between ref1 and ref2". Why is that? 

- 3: Should the "--splice" flag always be used on RNA seq data? Is it detrimental to use it if there is no splicing?

**Comment by tony-kuo** [comment]
Hello.

1.  Yes sorry, "[ref3_ngi_consensus.py](https://github.com/tony-kuo/eagle/blob/master/scripts/ref3_ngi_consensus.py)" is the script.
2. The chromosome names should reflect the subgenomes, for example "chr1A" and "chr1B". There should be some distinguishing factor in the name to separate subgenomes.
3. We usually use "--splice" for RNA as we expect splicing. If you're sure there is no splicing, then you can leave that out.

Hope this helped

**Comment by kenji-yt** [comment]
Great! Thank you so much.
